Mage.Client/src/main/java/org/mage/plugins/card/dl/sources/ScryfallImageSource.java

```diff
@@ -32,6 +33,14 @@ public enum ScryfallImageSource implements CardImageSource {
     private final Map<CardDownloadData, String> preparedUrls = new HashMap<>();
     private final int DOWNLOAD_TIMEOUT_MS = 100;
 
+    private Object readResolve() throws ObjectStreamException {
```

Comment:
It's a weird java thing - see https://docs.oracle.com/javase/8/docs/api/java/io/Serializable.html for details. I don't know for sure if it's actually needed, but take care before removing.

Reply:
Okay, this class doesn't extend Serializable so must not be needed here.

Reply:
Hmm. It require to support serialization of singleton classes (without that method each singleton object will be duplicated with new object). Xmage don't use serialization of server side objects, so it's safe from such errors (miss hidden fields/methods). But it can be important for potentially game replay or restore feature (old implementation used serialization to save and load game states from a file).

So it's better to keep such method in all enum style classes (singleton).